### PR TITLE
Add Min path sum in C code

### PR DESCRIPTION
**Pull request by Zuoxun**
Running results are as follows:
> 从左上角到右下角的最小路径和为 13
从左上角到右下角的最小路径和为 13        
从左上角到右下角的最小路径和为 13        
从左上角到右下角的最小路径和为 13

If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/c/chapter_dynamic_programming/min_path_sum.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ * Author: Zuoxun (845242523@qq.com)
+ */
+
+#include "../utils/common.h"
+
+int n, m;
+int grid_size, grid0_size;
+
+/* 最小路径和：暴力搜索 */
+int minPathSumDFS(int grid[][m], int i, int j) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/min_path_sum.c:17]: The formal parameter passing of this function is too awkward. Global variables are included in the parameter passing process, which is very inflexible.

Can be used as follows：

```c
int minPathSumDFS(int gridCols, int grid[][gridCols], int i, int j) {
```

File: codes/c/chapter_dynamic_programming/min_path_sum.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+    // 计算从左上角到 (i-1, j) 和 (i, j-1) 的最小路径代价
+    int left = minPathSumDFS(grid, i - 1, j);
+    int up = minPathSumDFS(grid, i, j - 1);
+    // 返回从左上角到 (i, j) 的最小路径代价
+    return min(left, up) != INT_MAX ? min(left, up) + grid[i][j] : INT_MAX;
+}
+
+/* 最小路径和：记忆化搜索 */
+int minPathSumDFSMem(int grid[][m], int mem[][m], int i, int j) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/min_path_sum.c:34]: Same as above

File: codes/c/chapter_dynamic_programming/min_path_sum.c [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    }
+    // 若行列索引越界，则返回 +∞ 代价
+    if (i < 0 || j < 0) {
+        return INT_MAX;
+    }
+    // 计算从左上角到 (i-1, j) 和 (i, j-1) 的最小路径代价
+    int left = minPathSumDFS(grid, i - 1, j);
+    int up = minPathSumDFS(grid, i, j - 1);
+    // 返回从左上角到 (i, j) 的最小路径代价
+    return min(left, up) != INT_MAX ? min(left, up) + grid[i][j] : INT_MAX;
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/min_path_sum.c:30]: `min` not declared

File: codes/c/chapter_dynamic_programming/min_path_sum.c [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    // 左边和上边单元格的最小路径代价
+    int left = minPathSumDFSMem(grid, mem, i - 1, j);
+    int up = minPathSumDFSMem(grid, mem, i, j - 1);
+    // 记录并返回左上角到 (i, j) 的最小路径代价
+    mem[i][j] = min(left, up) != INT_MAX ? min(left, up) + grid[i][j] : INT_MAX;
+    return mem[i][j];
+}
+
+/* 最小路径和：动态规划 */
+int minPathSumDP(int grid[][m]) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/min_path_sum.c]: Same as above

File: codes/c/chapter_dynamic_programming/min_path_sum.c [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    int n = grid_size, m = grid0_size;
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/min_path_sum.c]: Minimize the use of global variables
Here, the simultaneous existence of global variables and local variables m can easily cause misunderstandings.

File: codes/c/chapter_dynamic_programming/min_path_sum.c [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        for (int j = 1; j < m; j++) {
+            dp[i][j] = min(dp[i][j - 1], dp[i - 1][j]) + grid[i][j];
+        }
+    }
+    return dp[n - 1][m - 1];
+}
+
+/* 最小路径和：空间优化后的动态规划 */
+int minPathSumDPComp(int grid[][m]) {
+    int n = grid_size, m = grid0_size;
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/min_path_sum.c]: Same as above

File: codes/c/chapter_dynamic_programming/min_path_sum.c (hunk 1 above)
Comment (Zuoxun) [reply codes/c/chapter_dynamic_programming/min_path_sum.c]: Thank you, learned a lot, i've edited it following your suggestion.

File: codes/c/chapter_dynamic_programming/min_path_sum.c (hunk 3 above)
Comment (Zuoxun) [reply codes/c/chapter_dynamic_programming/min_path_sum.c:28]: I've declared function min at line 10 in the code, didn't you see it? How about refresh?

**Review by Gonglja — COMMENTED** [review]
Approve @krahets

File: codes/c/chapter_dynamic_programming/min_path_sum.c (hunk 3 above)
Comment (Gonglja) [reply codes/c/chapter_dynamic_programming/min_path_sum.c:28]: > I've declared function min at line 10 in the code, didn't you see it? How about refresh?

It didn’t happen before my review, but now it’s ok

**Review by krahets — APPROVED** [review]

**Pull request merged**